internal/rest/resources/control.go

```diff
@@ -210,20 +211,23 @@ func joinWithToken(state state.State, r *http.Request, req *internalTypes.Contro
 	}
 
 	// Get a client to the target address.
-	var lastErr error
+	var allErrors []error
 	var joinInfo *internalTypes.TokenResponse
 	for _, addr := range token.JoinAddresses {
 		url := api.NewURL().Scheme("https").Host(addr.String())
 
 		cert, err := shared.GetRemoteCertificate(url.String(), "")
 		if err != nil {
 			logger.Warn("Failed to get certificate of cluster member", logger.Ctx{"address": url.String(), "error": err})
```

Comment:
IIRC this is only logged as a warning because the join token contains multiple addresses and since we can try the other addresses too it's not considered to be an error that should be returned to the caller if one of the addresses cannot be reached as the join itself can still succeed with any of the remaining addresses.

Reply:
Right, which is why we're not returning an error here, we're simply adding it to the list of ``allErrors``, which is eventually joined, wrapped, and returned if the join failed for all the available addresses (`if joinInfo == nil`).

If the join action was successful, this warning and error will be discarded anyways, but if it failed, I'd say this bit of information is still more useful to a user rather than a confusing `Last error: %!w(<nil>)`, and if we failed to join through all the join addresses, it will still be useful to know what happened for each of them, letting the user know directly through the error we return, rather than having the user scour through the service logs.

Reply:
Thanks for adding the error assignment.